www/js/splash/startprefs.ts

```diff
@@ -0,0 +1,126 @@
+import { getAngularService } from "../angular-react-helper";
+import { storageGet, storageSet } from '../plugin/storage';
+import { logInfo, logDebug, displayErrorMsg } from '../plugin/logger';
+import { readIntroDone } from "../onboarding/onboardingHelper";
+
+// data collection consented protocol: string, represents the date on
+// which the consented protocol was approved by the IRB
+const DATA_COLLECTION_CONSENTED_PROTOCOL = 'data_collection_consented_protocol';
+
+let _req_consent;
+let _curr_consented;
+
+/**
+ * @function writes the consent document to native storage
+ * @returns Promise to execute the write to storage
+*/
+function writeConsentToNative() {
+  //note that this calls to the notification API, 
+  //so should not be called until we have notification permissions
+  //see https://github.com/e-mission/e-mission-docs/issues/1006
+  return window['cordova'].plugins.BEMDataCollection.markConsented(_req_consent);
+};
+
+/**
+ * @function marks consent in native storage, local storage, and local var
+ * @returns Promise for marking the consent in native and local storage
+ */
+export function markConsented() {
+  logInfo("changing consent from " +
+    _curr_consented + " -> " + JSON.stringify(_req_consent));
+  // mark in native storage
+  return readConsentState()
+    .then(writeConsentToNative)
+    .then(function (response) {
+      // mark in local storage
+      storageSet(DATA_COLLECTION_CONSENTED_PROTOCOL,
+        _req_consent);
+      // mark in local variable as well
+      _curr_consented = { ..._req_consent };
+    })
+    //check for reconsent
+    .then(readIntroDone)
+    .then((isIntroDone) => {
+      if(isIntroDone) {
+        console.debug("reconsent scenario - marked consent after intro done - registering pushnoify and storing device settings")
+        const PushNotify = getAngularService("PushNotify");
+        const StoreSeviceSettings = getAngularService("StoreDeviceSettings");
+        PushNotify.registerPush();
+        StoreSeviceSettings.storeDeviceSettings();
+      }
+    });
+};
+
+let _is_consented;
```

Comment:
Is this variable needed? It is assigned below, but I don't see it being used for anything

Reply:
You're right, that is assigned but never used, there were a lot of variables like that in the angular code, and I must have missed that this one isn't used anymore, thank you for pointing that out!